ifrs17/Import/Importers.ipynb

```diff
@@ -675,42 +694,74 @@
-                "\n                                .Select(x => datarow.Field<string>(x.ColumnName).CheckStringForExponentialAndConvertToDouble()).ToArray(),",
-                "\n                Name = hasNameColumn",
-                "\n                            ? datarow.Field<string>(nameof(YieldCurve.Name))",
-                "\n                            : default(string)",
-                "\n            };",
-                "\n            return committedYieldCurves.Contains(yieldCurve, YieldCurveComparer.Instance())",
-                "\n                    ? null",
-                "\n                    : yieldCurve;",
-                "\n            }",
-                "\n        ).WithTarget(workspace).ExecuteAsync(); ",
-                "\n    ",
+                "\n    var importLog = await Import.FromDataSet(dataSet).WithType<YieldCurve>((dataset, datarow) => { ",
+                "\n        var yieldCurve = new YieldCurve { ",
+                "\n            Currency = datarow.Field<string>(nameof(YieldCurve.Currency)),",
+                "\n            Year = args.Year,",
+                "\n            Month = args.Month, ",
+                "\n            Scenario = args.Scenario,",
+                "\n            Values = datarow.Table.Columns.Where(c => c.ColumnName.StartsWith(nameof(YieldCurve.Values))).OrderBy(c => c.ColumnName.Length).ThenBy(c => c.ColumnName)",
+                "\n                            .Select(x => datarow.Field<string>(x.ColumnName).CheckStringForExponentialAndConvertToDouble()).ToArray(),",
+                "\n            Name = hasNameColumn ? datarow.Field<string>(nameof(YieldCurve.Name)) : default(string)",
+                "\n        };",
+                "\n        return committedYieldCurves.Contains(yieldCurve, YieldCurveComparer.Instance()) ? null : yieldCurve;",
+                "\n    }).WithTarget(workspace).ExecuteAsync(); ",
+                "\n",
                 "\n    if(importLog.Errors.Any()) return Activity.Finish().Merge(importLog); ",
-                "\n    ",
-                "\n    var toCommitYieldCurves = await workspace.Query<YieldCurve>().ToArrayAsync();",
-                "\n    if (!toCommitYieldCurves.Any()){",
+                "\n    var toCommitYieldCurves  = await workspace.Query<YieldCurve>().ToArrayAsync();",
+                "\n    if (!toCommitYieldCurves.Any()) {",
                 "\n        ApplicationMessage.Log(Warning.VariablesAlreadyImported);   ",
                 "\n        return Activity.Finish().Merge(importLog);",
                 "\n    }",
-                "\n    ",
-                "\n    await DataSource.UpdateAsync<YieldCurve>(toCommitYieldCurves);",
-                "\n    await DataSource.CommitAsync();",
+                "\n",
+                "\n    PartitionByReportingNodeAndPeriod[] targetPartitions;",
```

Comment:
I would try to make this method easier to read by extracting some methods. 
For example these lines to set the targetPartitions could be a method. 
Moreover, other importers require a similar functionality. For example Cashflow format uses a function call getAllArgs which (I think) serves a similar purpose. I suggest to unify the methods (possibly avoiding some code repetition). It is possible this methos would require the importFormat as an input to discriminate specific logic, but I think this is a cost we can pay to improve readability.

Reply:
To increase readability I would also use the same variable names across different import format. 
eg in cashflow importer we loop over allArgs, here we loop over targetPartitions, but the meaning is exactly the same.

Reply:
done! GetAllArgs is now used for both the transactional data and the yield curve importers. Also, this means that the variable names are now consistent.